odxtools/units.py

```diff
@@ -0,0 +1,290 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2021 MBition GmbH
+
+from dataclasses import dataclass, field
+from typing import List, Literal, Optional
+
+from .nameditemlist import NamedItemList
+from .utils import read_description_from_odx
+
+UnitGroupCategory = Literal["COUNTRY", "EQUIV-UNITS"]
+
+
+@dataclass
+class PhysicalDimension:
+    """A physical dimension is a formal definition of a unit.
+
+    It consists of the exponents for the SI units:
+
+    | Symbol | Name     | Quantity                  | Property                 |
+    |  ---   | ---      | ---                       | ---                      |
+    | s      | second   | time                      | `time_exp`               |
+    | m      | metre    | length                    | `length_exp`             |
+    | kg     | kilogram | mass                      | `mass_exp`               |
+    | A      | ampere   | electric current          | `current_exp`            |
+    | K      | kelvin   | thermodynamic temperature | `temperature_exp`        |
+    | mol    | mole     | amount of substance       | `molar_amount_exp`       |
+    | cd     | candela  | luminous intensity        | `luminous_intensity_exp` |
+
+    (The first three columns are from https://en.wikipedia.org/wiki/International_System_of_Units.)
+
+
+    Examples
+    --------
+
+    The unit `m/s` (or `m**1 * s**(-1)`) can be represented as
+    ```
+    PhysicalDimension(
+        id="velocity",
+        short_name="metre_per_second",
+        length_exp=1,
+        time_exp=-1
+    )
+    ```
+    """
+    id: str
+    short_name: str
+    oid: Optional[str] = None
+    long_name: Optional[str] = None
+    description: Optional[str] = None
+    length_exp: int = 0
+    mass_exp: int = 0
+    time_exp: int = 0
+    current_exp: int = 0
+    temperature_exp: int = 0
+    molar_amount_exp: int = 0
+    luminous_intensity_exp: int = 0
+
+
+@dataclass
+class Unit:
+    """
+    A minimal unit just consists of an ID, short name and a display name:
+
+    ```
+    Unit(
+        id="kilometre",
+        short_name="kilometre",
+        display_name="km"
+    )
+    ```
+
+    Additionally, a unit may reference a SI unit (`.physical_dimension`)
```

Comment:
This is not a big deal, since I frequently observe native speakers getting this wrong: I guess this should be "an SI unit". Like any other natural language English has weird quirks, in this case it's that if something is *pronounced* as starting with a vowel, "an" is used, if it is not, "a" will win out even if the word in question starts with a vowel. Examples: "This is an SI unit" but "a unit may reference".

Reply:
Thanks!